pennylane/templates/embeddings/amplitude.py

```diff
@@ -171,11 +172,11 @@ def circuit(f=None):
     #############
     # Input checks
 
+    wires = Wires(wires)
```

Comment:
Is there any significance in changing the order and having the wire creation here? (saw this was done later too)

Reply:
Only that a potential error for incorrect wires gets thrown as early as possible. But other than that, once could move this line to just before the definition of  ``wire_sequence``...